VerticalSlice2D/Assets/Scripts/Dialogue/DialogueLogic.cs

```diff
@@ -0,0 +1,39 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using TMPro;
+
+public class DialogueLogic : MonoBehaviour
+{
+    private bool isActive;
+    [SerializeField] private GameObject dialogueBoxUI;
+    [SerializeField] private TextMeshProUGUI dialogueUIText;
+    [SerializeField] private Dialogue dialogueObj;
+    [SerializeField] private float talkingSpeed, pauzeSpeed;
+
+    private void Update() 
+    {
+        if (Input.GetKeyDown(KeyCode.Space) && !isActive) 
+        {
+            dialogueBoxUI.SetActive(true);
+            StartCoroutine(StartDialogue());
+        }
+    }
+    private IEnumerator StartDialogue() 
+    {
+        isActive = true;
+        dialogueUIText.text = string.Empty;
+        foreach (char letter in dialogueObj.dialogueMessage) 
+        {
+            dialogueUIText.text += letter;
+            if(!char.IsWhiteSpace(letter)){
+                FindObjectOfType<AudioManager>().Play("DialogueVoice");
```

Comment:
In plaats van elke keer `FindObjectOfType` te doen het 1 keer op te slaan in een variabel boven de `foreach` loop